usecases/utilities_test.go

```diff
@@ -0,0 +1,377 @@
+		Unit:    "testUnit",
+		Version: "testVersion",
+	}
+	return form
+}
+
+type brokenMockForm struct {
+	XMLName xml.Name  `json:"-" xml:"testName"`
+	Value   complex64 `json:"value" xml:"value"`
+	Unit    string    `json:"unit" xml:"unit"`
+	Version string    `json:"version" xml:"version"`
+}
+
+// NewForm creates a new form
+func (f brokenMockForm) NewForm() forms.Form {
+	f.Version = "testVersion"
+	return f
+}
+
+// FormVersion returns the version of the form
+func (f brokenMockForm) FormVersion() string {
+	return f.Version
+}
+
+// Returns a form containing complex numbers, which xml and json can't marshal
+func createBrokenTestForm() (f brokenMockForm) {
+	form := brokenMockForm{
+		XMLName: xml.Name{},
+		Value:   complex(1, 2),
+		Unit:    "testUnit",
+		Version: "testVersion",
+	}
+	return form
+}
+
+type packParams struct {
+	contentType   string
+	form          forms.Form
+	expectedError bool
+	testCase      string
+}
+
+// Returns an error containing a list values who was missing/wrong
+func assurePackData(byteArr []byte, contentType string, expectedError bool) (err error) {
+	data := string(byteArr)
+	if contentType == "application/xml" {
+		missingData := []string{}
+		correctName := strings.Contains(data, "<testName>")
+		if correctName != true {
+			missingData = append(missingData, "XMLName")
+		}
+		if expectedError == false {
+			correctValue := strings.Contains(data, "<value>123</value>")
+			if correctValue != true {
+				missingData = append(missingData, "Value")
+			}
+		} else {
+			correctValue := strings.Contains(data, "<value>(1+2i)</value>")
+			if correctValue != true {
+				missingData = append(missingData, "Value")
+			}
+		}
+		correctUnit := strings.Contains(data, "<unit>testUnit</unit>")
+		if correctUnit != true {
+			missingData = append(missingData, "Unit")
+		}
+		correctVersion := strings.Contains(data, "<version>testVersion</version>")
+		if correctVersion != true {
+			missingData = append(missingData, "Version")
+		}
+		if len(missingData) != 0 {
+			return fmt.Errorf("missing data: %s", missingData)
+		}
+	} else {
+		missingData := []string{}
+		if expectedError == false {
+			correctValue := strings.Contains(data, `"value": 123`)
+			if correctValue != true {
+				missingData = append(missingData, "Value")
+			}
+		} else {
+			correctValue := strings.Contains(data, `"value": (1+2i)`)
+			if correctValue != true {
+				missingData = append(missingData, "Value")
+			}
+		}
+		correctUnit := strings.Contains(data, `"unit": "testUnit"`)
+		if correctUnit != true {
+			missingData = append(missingData, "Unit")
+		}
+		correctVersion := strings.Contains(data, `"version": "testVersion"`)
+		if correctVersion != true {
+			missingData = append(missingData, "Version")
+		}
+		if len(missingData) != 0 {
+			return fmt.Errorf("missing data: %s", missingData)
+		}
+	}
+	return nil
+}
+
+func TestPack(t *testing.T) {
+	params := []packParams{
+		{"application/xml", createTestForm(), false, "Best case, xml"},
```

Comment:
Maybe add a "testing function" to this table and then you can move a single branch of testing code (from assurePackData) to this table.